rules/order/src/Rector/Class_/OrderMethodsByVisibilityRector.php

```diff
@@ -42,22 +44,6 @@ final class OrderMethodsByVisibilityRector extends AbstractRector
         'tearDown',
     ];
 
-    /**
-     * @var StmtOrder
-     */
-    private $stmtOrder;
-
-    /**
-     * @var StmtVisibilitySorter
-     */
-    private $stmtVisibilitySorter;
-
-    public function __construct(StmtOrder $stmtOrder, StmtVisibilitySorter $stmtVisibilitySorter)
-    {
-        $this->stmtOrder = $stmtOrder;
-        $this->stmtVisibilitySorter = $stmtVisibilitySorter;
-    }
```

Comment:
@dobryy These 2 services are repeated over and over again, so it makes sense to decouple `Abstract...Rector`, that has them both.

Now it's easy to add common method like  `hasOrderChanged()` ;)

Reply:
Usually I don't use Abstract classes when it's all about DI, if this is how you do it in Rector, I don't mind to follow this rule (PHPStan? :D). Now it makes sense to use Abstract class because there is new method that is required everywhere.

Reply:
Sure :) If you can distile PHPStan rule from this, go for it :+1: 

It's usually about common services/methods/logics. Same goes for PHPUnitAbstractRector.